src/components/TextField/TextField.vue

```diff
@@ -0,0 +1,405 @@
+			<div class="text-field__icon text-field__icon--leading">
+				<!-- Leading material design icon in the text field, set the size to 18 -->
+				<slot />
+			</div>
+
+			<!-- Success icon -->
+			<div v-if="success" class="text-field__icon text-field__icon--trailing">
+				<Check :size="18" />
+			</div>
+
+			<!-- clear text button -->
+			<Button v-if="canClear"
+				type="tertiary-no-background"
+				class="text-field__clear-button"
+				@click="clearText">
+				<Close slot="icon"
+					:size="20" />
+			</Button>
+		</div>
+	</div>
+</template>
+
+<script>
+import Button from '../Button/index.js'
+import Close from 'vue-material-design-icons/Close'
+import Check from 'vue-material-design-icons/Check'
+import GenRandomId from '../../utils/GenRandomId.js'
+
+export default {
+	name: 'TextField',
+
+	components: {
+		Button,
+		Close,
+		Check,
+	},
+
+	props: {
+		/**
+		 * The value of the input field
+		 */
+		value: {
+			type: String,
+			required: true,
+		},
+
+		/**
+		 * The hidden input label for accessibility purposes. This will also
+		 * be used as a placeholder unless the placeholder prop is populated
+		 * with a different string.
+		 */
+		label: {
+			type: String,
+			default: undefined,
+		},
+
+		/**
+		 * Pass in true if you want to use an external label. This is useful
+		 * if you need a label that looks different from the one provided by
+		 * this component
+		 */
+		labelOutside: {
+			type: Boolean,
+			default: false,
+		},
+
+		/**
+		 * We normally have the lable hidden visually and use it for
+		 * accessibility only. If you want to have the label visible just above
+		 * the input field pass in true to this prop.
+		 */
+		labelVisible: {
+			type: Boolean,
+			default: false,
+		},
+
+		/**
+		 * The placeholder of the input. This defaults as the string that's
+		 * passed into the label prop. In order to remove the placeholder,
+		 * pass in an empty string.
+		 */
+		placeholder: {
+			type: String,
+			default: undefined,
+		},
+
+		/**
+		 * Controls whether to display the clear button or not. Since the
+		 * parent component will have to store the value of this text input,
+		 * once clear button is pressed, there's no change in the value of
+		 * the text input. Instead, a 'clear' event is sent to the parent
+		 * component.
+		 */
+		canClear: {
+			type: Boolean,
+			default: false,
+		},
+
+		/**
+		 * Toggles the success state of the component. Adds a checkmark icon.
+		 * this cannot be used together with canClear.
+		 */
+		success: {
+			type: Boolean,
+			default: false,
+		},
+	},
+
+	computed: {
+		inputName() {
+			return 'input' + GenRandomId()
+		},
+
+		hasLeadingIcon() {
+			return this.$slots.default
+		},
+
+		hasTrailingIcon() {
+			return this.success
+		},
+
+		hasPlaceholder() {
+			return this.placeholder !== '' && this.placeholder !== undefined
+		},
+
+		computedPlaceholder() {
+			if (this.labelVisible) {
+				return this.placeholder ? this.placeholder : ''
```

Comment:
```suggestion
				return this.hasPlaceholder ? this.placeholder : ''
```
To be consistent with below?